File: PhoneNumberChecker/Models/NumberValidationDTO.cs
using System;
using System.Globalization;
using PhoneNumbers;

namespace PhoneNumberChecker.Models
{
    public partial class NumberValidationDTO
    {
        public Guid Id { get; set; }
        public ulong PhoneNumber { get; set; }
        public string? IntlFormat { get; set; }
        public bool IsValid { get; set; }
        public bool IsPossible { get; set; }
        public PhoneNumberType NumberType { get; set; }
        public string? Region { get; set; }

        // To be used by the frontend to set selected rows
        public bool IsSelected => false;

        // Format text based on run environment
        public string FormattedNumberType => Thread.CurrentThread.CurrentCulture.TextInfo.ToTitleCase(
                string
                .Join(" ", (Enum.GetName(NumberType)?.ToLower() ?? "")
                            .Split("_")
                    )
            );
    }
}

File: Client/src/models/NumberValidationDTO.ts
export enum PhoneNumberType {
  FIXED_LINE = 0,
  MOBILE = 1,
  FIXED_LINE_OR_MOBILE = 2,
  TOLL_FREE = 3,
  PREMIUM_RATE = 4,
  SHARED_COST = 5,
  VOIP = 6,
  PERSONAL_NUMBER = 7,
  PAGER = 8,
  UAN = 9,
  VOICEMAIL = 10,
  UNKNOWN = 11,
}

export interface NumberValidationDTO {
  id: string;
  phoneNumber: number;
  intlFormat: string | null;
  isValid: boolean;
  isPossible: boolean;
  numberType: PhoneNumberType;
  region: string | null;
  formattedNumberType: string;
  isSelected: boolean;
}
